### ISSUE TITLE
Cannot disable Container Cap anymore

### ISSUE THREAD

**Issue by jonathanperret** [jenkinsci/docker-plugin#318]
The documentation at https://wiki.jenkins-ci.org/display/JENKINS/Docker+Plugin says:

> **Container Cap** The maximum number of containers that this provider is allowed to have running, defaults to blank which disables the limit

However, since @KostyaSha committed https://github.com/jenkinsci/docker-plugin/commit/bde53df15d84e746dba3a40fcaa050bf60260ab1 , this is no longer true: an empty "Container Cap" field is saved as a zero value, and interpreted as such, i.e. no containers can ever start.

**Comment by KostyaSha** [comment]
You can't run 32768 containers or what? What is your issue?

**Comment by KostyaSha** [comment]
Documentation updated

**Comment by jonathanperret** [comment]
Sorry, I should have been more explicit in the bug report. Here goes:
* Configure a "Docker cloud" with "Container cap" set to 0.
* Try to start a job bound to that cloud provider.

Observed behavior:
* The job does not start. In the Jenkins logs, the following message appears: "Not Provisioning 'xxx'; Server 'yyy' full with '0' container(s)"

Expected behavior:
* The job starts.

I hope this is more useful.
Regards.

**Comment by KostyaSha** [comment]
You should specify explicitly container limits, previous zero default was 32768.

**Comment by magnayn** [comment]
I did warn about this at the time.

You now basically have to set a sufficiently high value, with no possible
option for "I do not wish to set a container cap".

**Comment by KostyaSha** [comment]
While jenkins has constant issues with provisioning it bad to have unlimited limits.

**Comment by jonathanperret** [comment]
If removing the option to not set a container cap was deliberate, I won't argue. I just reported this issue in case it had been an oversight.

May I suggest however that the documentation be updated to say something else than "0 disables capacity", which might lead users to expect the behavior I described? Perhaps something like "If this is set to 0, no containers will ever be started."?

By the way, although this is pure nitpicking, INT_MAX is 2<sup>31</sup>-1 (as per http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#MAX_VALUE) and not 32768 :smile:
